src/Promenade.Controllers/LocationsController.cs

```diff
@@ -186,8 +190,49 @@ public async Task<IActionResult> Locations(string locationStub, string featureSt
                     LocationFeatures = new List<LocationsFeaturesViewModel>(),
                     Location = await _locationService.GetLocationByStubAsync(locationStub)
                 };
+                if (viewModel.Location.PreFeatureSegmentId != null)
+                {
+                    var segment = await _segmentService.GetSegmentById(viewModel.Location.PreFeatureSegmentId.Value);
+                    if (segment.IsActive)
```

Comment:
Business logic (e.g. if the segment is active and is scheduled to be displayed) should happen in the service call and not the controller.